### PR TITLE
refactor: es6, forced style linting, nodemon for autorestart of server

### PR DESCRIPTION
**Pull request by AVVS**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,12 +19,19 @@
...
+    "eslint": "^3.1.1",
+    "eslint-config-airbnb-base": "^4.0.2",
+    "eslint-plugin-import": "^1.11.1",
+    "eslint-plugin-promise": "^2.0.0",
+    "nodemon": "^1.9.2"
   },
   "scripts": {
     "test": "npm run lint",
-    "lint": "jshint --exclude-path .gitignore ."
+    "lint": "eslint ./lib",
```
Comment (jozsi) [inline package.json:32]: This wouldn't cover the root folder, which currently consists of a minimal `index.js`. If we stick to use that only as an entry point, I guess there's no linting to go wrong...

Comment (AVVS) [inline package.json:32]: or we can actually remove index and just add a proper main

File: lib/mitm/handlers/autoFlick.js [hunk 2]
```diff
@@ -1,47 +1,48 @@
+const debug = require('debug')('mitm:autoFlick');
```
Comment (rubenvereecken) [inline lib/mitm/handlers/autoFlick.js:1]: I like this

File: lib/mitm/handlers/autoFlick.js [hunk 3]
```diff
@@ -1,47 +1,48 @@
...
-          var fortLocation = {
-            latitude: fort.latitude,
-            longitude: fort.longitude,
-          };
-          var distance = haversine(app.locals.player.location, fortLocation);
-          if (distance < 0.05) {
-            return fort;
-          }
+module.exports = (mitmServer, app) => {
+  mitmServer.addResponseHandler('GetMapObjects', (data) => safeProcess(() => {
```
Comment (rubenvereecken) [inline lib/mitm/handlers/autoFlick.js:10]: and this

File: lib/mitm/handlers/betterThrow.js [hunk 4]
```diff
@@ -2,7 +2,8 @@ const rand = require('randgen');
 const mitmConfig = require('../../config').mitm;
 
 module.exports = mitmServer => {
-  //TODO Also add the NormalizedHitPosition as soon as content length is fixed
+  // TODO Also add the NormalizedHitPosition as soon as content length is fixed
```
Comment (rubenvereecken) [inline lib/mitm/handlers/betterThrow.js:5]: I format it like that because otherwise my vim doesn't highlight it :)

Comment (AVVS) [inline lib/mitm/handlers/betterThrow.js:5]: yeah well, linter complains :)

File: lib/player.js [hunk 5]
```diff
@@ -1,6 +1,5 @@
 class Player {
-  contructor(proto, authInfo) {
-    if (!proto) proto = {};
+  contructor(proto = {}, authInfo) {
```
Comment (rubenvereecken) [inline lib/player.js:2]: Hm what language allows a defaulted first parameter without a second

File: lib/server.js [hunk 6]
```diff
@@ -1,17 +1,50 @@
...
+  };
+
+  process.once('SIGUSR2', gracefulStop('SIGUSR2'));
+  process.once('SIGINT', gracefulStop('SIGINT'));
+}
 
 setupExpress(app);
 setupMitm(app);
 
+if (!module.parent) {
```
Comment (rubenvereecken) [inline lib/server.js:44]: nice

**Comment by rubenvereecken** [comment]
:tada:

**Pull request merged**